src/client/index.ts

```diff
@@ -18,111 +79,268 @@ export interface NotificationOptions {
  *
- * This will not fire in cases of Notification Buttons being clicked.  See {@link NotificationButtonClickedEvent}.
+ * This type includes a generic type argument, should applications wish to define their own interface for action
+ * results. See {@link NotificationActionResult} for details.
  *
  * @event
  */
-export interface NotificationClickedEvent {
-    type: 'notification-clicked';
-    notification: Notification;
+export interface NotificationActionEvent<T = {}> {
+    type: 'notification-action';
+
+    /**
+     * The notification that created this action
+     */
+    notification: Readonly<Notification>;
+
+    /**
+     * Indicates what triggered this action.
+     *
+     * Note that the `programmatic` trigger is not yet implemented.
+     */
+    trigger: ActionTrigger;
+
+    /**
+     * The control whose interaction resulted in this action being raised. Will only be present when {@link trigger} is
+     * {@link ActionTrigger.CONTROL}.
+     *
+     * Future versions of the service will add additional controls beyond buttons, and interactions with these new
+     * control types will also come through this one event type. For best forward-compatibility, applications should
+     * always check the `type` property of this control, and not assume that the type will always be `'button'`.
+     *
+     * This field is marked optional as future versions of the service will also include alternate methods of raising
+     * `notification-action` events that do not originate from a button or other control.
+     *
+     * When present, the object here will always be strictly equal to one of the control definitions within
+     * `notification`. This means `indexOf` checks and other equality checks can be performed on this field if
+     * required, such as:
+     *
+     * ```ts
+     * function onNotificationAction(event: NotificationActionEvent): void {
+     *     if (event.control && event.control.type === 'button') {
+     *         const butttonIndex = event.notification.buttons.indexOf(event.control);
+     *
+     *         // Handle button click
+     *         // . . .
+     *     }
+     * }
+     * ```
+     */
+    control?: Readonly<ControlOptions>;
+
+    /**
+     * Application-defined metadata that this event is passing back to the application.
+     *
+     * A `notification-action` event is only raised if the {@link NotificationOptions|notification definition} includes
+     * a payload on the actions of one or more controls (and if the user then performs the corresponding action).
+     *
+     * See the comment on the {@link NotificationActionEvent} type for an example of buttons that do and don't raise
+     * actions.
+     */
+    result: NotificationActionResult<T>;
 }
 
 /**
- * Event fired whenever the Notification has been closed.
+ * Lists the different triggers for a notification {@link Actions|action}. Each action that is triggered will result in
```

Comment:
Oh god! Should we be saying "trigger action" rather than "raised action" everywhere?!

Also, would go with "created the notification" rather than "raised the notification"

Reply:
In the context we're using 'raised' in the docs, I think 'raised' is a better fit...

Changed 'created the notification'.

(FYI, this type has now been moved to `src/client/actions.ts`)

Reply:
"Lists the different triggers that can raise an {@link Actions|action}. Each action that is raised will result in..." ?